### ISSUE TITLE
BadRequest,Filter failure: LOT_SIZE

### ISSUE THREAD

**Issue by heavymanto** [JKorf/Binance.Net#702]
hi, i try to place an order but it respon thi error LOT_SIZE.  I read that there are minimums and maximums for symbol but I don't understand where I find this data

CODE
```
var _operation = (Binance.Net.Enums.OrderSide)(int)operation;
string _symbol = ((IExchangeClient)_client).GetSymbolName(_valutaBase, _valutaQuote);
var order = _client.Spot.Order.PlaceTestOrderAsync(_symbol, _operation, Binance.Net.Enums.OrderType.Market, amount);

```

DATA/LOG OF ORDER RESULT
_volumeTradableRround:465,939801
 ApiClient_Binance.PlaceOrder --> BadRequest,Filter failure: LOT_SIZE

**Comment by heavymanto** [comment]
maybe I understand, the problem is the steSize, which must undergo the following equation

![image](https://user-images.githubusercontent.com/6449169/116008817-c8020900-a616-11eb-98af-eada37a10cd8.png)

specifically the symbol has the stepSize to 1.0 which I think means to round to the whole part.
![image](https://user-images.githubusercontent.com/6449169/116008810-bd477400-a616-11eb-9336-b34bce2f0439.png)

**Comment by heavymanto** [comment]
I have found BinanceHelpers.ClampQuantity and BinanceHelpers.ClampPrice but it not use in PlaceOrder mehod. 
Why not integrate it?

**Comment by heavymanto** [comment]
SOLVED with this code. _outputQuantity is correct round amount

` var _exchangeInfo = _client.Spot.System.GetExchangeInfo();
            var _symbolData = _exchangeInfo.Data.Symbols.SingleOrDefault(s => string.Equals(s.Name, _symbol, StringComparison.CurrentCultureIgnoreCase));

            var minQty = _symbolData.LotSizeFilter?.MinQuantity;
            var maxQty = _symbolData.LotSizeFilter?.MaxQuantity;
            var stepSize = _symbolData.LotSizeFilter?.StepSize;

            var _outputQuantity = BinanceHelpers.ClampQuantity(minQty.Value, maxQty!.Value, stepSize!.Value, amount);`

**Comment by kamranhuseyns** [comment]
Hi my friend, I have the same problem. Do you run this code every time or if you get an error, do you run it then? thanks.

**Comment by heavymanto** [comment]
Whenever. Although it is written a bit complex, it is basically a rounding to one decimal place. It is always needed. Since using it I have not had a ko

**Comment by kamranhuseyns** [comment]
If there is always a need, do you know why Binance.Net is set to None by default?

if None it doesn't look at other codes.
![2022-02-01_121754](https://user-images.githubusercontent.com/63323757/151942297-4d5b6a0d-185d-4ae0-98e6-b5f6f2e848ac.png)
![2022-02-01_122211](https://user-images.githubusercontent.com/63323757/151942560-4b567048-a3b1-4102-b246-b54b4110d50f.png)

**Comment by heavymanto** [comment]
In my opinion it's a bug. The code is very complex, perhaps too much. I wrote it this way and it always works

![image](https://user-images.githubusercontent.com/6449169/151944116-320214ab-bf84-485c-9872-16a55c3ad0f6.png)

**Comment by kamranhuseyns** [comment]
Thank you very much!

**Comment by MattiaaFanti** [comment]
> 

hai capito come risolvere dunque? ho provato la tua soluzione, forse sbagliando ma non mi va..
Sto semplicemente facendo il parsing del prezzo di doge dalle api di binance.
dunque stabilendo la quantità= miei busd / prezzo doge
ricevo sempre l'errore lot size failure .. Anche con un toString("n5" o math round 5 nisba...

**Comment by heavymanto** [comment]
> hai capito come risolvere dunque? ho provato la tua soluzione, forse sbagliando ma non mi va..
> Sto semplicemente facendo il parsing del prezzo di doge dalle api di binance.
> dunque stabilendo la quantità= miei busd / prezzo doge
> ricevo sempre l'errore lot size failure .. Anche con un toString("n5" o math round 5 nisba...
> 

Ciao il mio codice funziona perfettamente. Usa la funzione già implementata nella libreria che calcola già il lotto minimo e passalo come paramento all'ordine.

`var _outputQuantity = BinanceHelpers.ClampQuantity(minQty.Value, maxQty!.Value, stepSize!.Value, amountRound);`

Dopo molte prove mi sono accorto che Binance non accetta ordini minori di circa 30$ a prescindere dalla valuta. Io solitamente non faccio mai ordini inferiori ai 50$ e non ho mai avuto problemi, altrimenti è probabile che li rifiuta. 

Per fare i primi esperimenti con pochi dollari ho usato Kucoin che accetta cifre molto basse.

----------------------------------------------------------------------------
TRANSLATE
> I'm simply parsing the price of doge from the binance bees.
> thus establishing the quantity = my busd / doge price


Use the function already implemented in the library that already calculates the minimum lot and pass it as a parameter to the order.

`var _outputQuantity = BinanceHelpers.ClampQuantity(minQty.Value, maxQty!.Value, stepSize!.Value, amountRound);`

After many tests, I realized that Binance does not accept orders of less than around $30 regardless of currency. I usually never place orders under $50 and have never had a problem, otherwise I'm likely to reject them.

To make my first experiments with a few dollars, I used Kucoin which accepts very low amounts.
